modules/doc/content/ncrc/applications/ncrc_develop.md.template

```diff
@@ -0,0 +1,159 @@
+
+## Clone {{ApplicationName}} id=clone
+
+Instructions for obtaining a proper clone of {{ApplicationName}} can be found at
+[NCRC/{{ApplicationName}}](ncrc/applications/ncrc_level4_{{ApplicationLower}}.md).
+
+## Support Libraries
+
+Most MOOSE-based applications require additional support libraries available in the form of
+submodules. Obtaining these submodules is a common step everyone must follow:
+
+```bash
+cd ~/projects/{{binary}}
+git submodule update --init
+```
+
+Depending on your preferred environment (Personal Machine or [!ac](HPC)) you will need to install
+Conda packages or build the requirements from source.
+
+- +Personal Machine (Conda)+:
+
+  ```bash
+  mamba create -n moose moose-tools moose-libmesh
+  ```
+
+- +[!ac](INL) [!ac](HPC) Sawtooth or Lemhi+:
+
+  It will be necessary to build PETSc and libMesh using the supplied build scripts in
+  `moose/scripts`. The time it takes to build both PETSc and libMesh can be hours:
+
+  ```bash
+  module load use.moose moose-dev    # a reminder to have these loaded
+  export MOOSE_JOBS=6 METHODS=opt    # a reminder to set these
+  cd ~/projects/{{binary}}/moose
+  scripts/update_and_rebuild_petsc.sh
+  scripts/update_and_rebuild_libmesh.sh
+  ```
+
+  !alert tip title=MOOSE_JOBS? METHODS?
+  Go to [MOOSE Build System](application_development/build_system.md) for more information on
+  available methods. MOOSE_JOBS controls how many cores to use during the use of our scripts.
+
+## Build {{ApplicationName}} id=build
+
+Building {{ApplicationName}} starts with entering the directory, bringing in any updates from our
+upstream remote and finally running `make`. Before we begin, +burn the next two warnings into your
+brain+:
+
+!alert warning title=You performed a submodule update?
+`Then you should perform a Conda update` (if you are using Conda).
+
+!alert warning title=You updated Conda?
+`Then you should perform a submodule update`.
+
+The symmetry importance comes from our Conda packages being version controlled by and within the
+MOOSE repository. So any updates to MOOSE (the `{{binary}}/moose` submodule) requires you to stay on
+top of any Conda updates, or rebuilding PETSc and libMesh.
+
+If this is your first time following these instructions there should be no updates (you have a brand
+new fork/clone/conda install). We will perform an update to familiarize yourself with the process:
+
+```bash
+cd ~/projects/{{binary}}
+git fetch upstream
+git branch   # take note of the results
+```
+
+`git fetch upstream` updates your local references with that of the remote's. It does not 'modify'
+any source files. It will return nothing if there was nothing new to fetch on the `upstream` remote.
+
+`git branch` reports what branch you have checked out. It is impossible to write a document for
+every application's branch naming schemes, so whatever was reported with `git branch`, use that for
+the next command.
```

Comment:
This probably needs to be clarified. `git branch` will in general return multiple branch names (remember that this section is also read for people *updating* the app). I'd delete "git branch" from your steps above and instead give an explanation of which branch to use:

```
git rebase -i upstream/somebranchname
```
where `somebranchname` is the default branch of {{ApplicationName}}, which can be found using
```
git remote show upstream | grep "HEAD branch" | sed 's/.*: //'
```

Reply:
Yeah, I was wrong about this... The correct one-liner is:
```bash
git rev-parse --abbrev-ref HEAD
```